bin/Model.js

```diff
@@ -57,48 +54,41 @@ module.exports = class Model {
 
                             this.addWeaponData(profile, selection);
                             break;
-                        case "abilities":                                    
+                        case "abilities":
                             this.addAbilityData(profile);
                             break;
                     }
-            
+
             if (selection.selections && selection.selections[0] !== "")
                 this.handleSelectionDataRecursive(selection.selections);
         }
     }
 
-    setWeapons(weaponArray) {
-        this.weapons = weaponArray;
-    }
-
     add(model) {

```

Comment:
This needs to go back in. If you load a few different rosters it will eventually error on this line. I added back in my PR and works again.

Reply:
oh, yes, it is used in `Unit.checkWeapons()`.

Since it does not trigger in my tests, can you provide a test case/ros file for it ?

Reply:
[750 - necron.zip](https://github.com/Yellow-Man-Code/BS2TTS/files/7982995/750.-.necron.zip)

here you go